docs/Adding_a_Service.rst

```diff
@@ -0,0 +1,100 @@
+Adding a New Service
+====================
+
+When adding a new service (i.e. directory in this repo), it is helpful to
+understand both:
+
+  - How PyTest is used in this application.
+  - What our naming conventions are.
+
+Recap of PyTest Services
+------------------------
+
+The expected workflow for a pytest service is:
+
+-   ``pytest`` runs, and produces a results*.json file:
+      - fetch any needed data
+      - test that data
+      - for failures, see if the failure is expected
+-   ``generate_service_report`` runs, and transforms the json file into a compact
+    listing of errors
+-   ``create_bugs`` [#create_bugs]_ runs, and takes action as appropriate (e.g.
+    open a bug).
+
+The expected use cases are:
+
+  - As an system operator, I can execute pytest interactively from the command
+    line to check parts of the system-under-test.
+
+  - As a test developer, I can specify how the tests should be run inside a
+    docker container, include taking automatic actions.
+
+  - As a test operator, I can invoke docker instances from automation to perform
+    the tests.
+
+  - As an interested party, I can easily modify parts of this code for use in my
+    environment.
+
+Overview
+--------
+
+To add a new service, you'll be writing or modifiying at least the following
+files:
+
++----------------------------------------------------------+--------+------------------------------------------------------------+
+|file                                                      | action | description                                                |
++==========================================================+========+============================================================+
+| pytest-services/{NEW_SERVICE}                            | create | directory to hold the service specific files               |
++----------------------------------------------------------+--------+------------------------------------------------------------+
+| pytest-services/{NEW_SERVICE}/test_*.py                  | write  | your test cases                                            |
++----------------------------------------------------------+--------+------------------------------------------------------------+
+| pytest-services/{NEW_SERVICE}/resources.py               | write  | Code to fetch & format the data you need to test           |
++----------------------------------------------------------+--------+------------------------------------------------------------+
+| pytest-services/{NEW_SERVICE}/helpers.py                 | write  | Code to simplify writing tests                             |
```

Comment:
May not need to be mentioned here, but this is also done so that we can have doctests for the bulk of the test code logic. "Test the tests!" :tm:  @g-k

Reply:
Yes, that should be mentioned somewhere here. I have notes on which files typically get them to add in. That was part of what helped me see the differences.